scale/ingest/views.py

```diff
@@ -339,6 +340,26 @@ def _create_v6(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers=dict(location=scan_url))
 
 
+
+class CancelScansView(GenericAPIView):
+    """This view is the endpoint for canceling a scan in progress."""
+    queryset = Scan.objects.all()
+    
+    def get_serializer_class(self):
+        """Returns the appropriate serializer based off the requests version of the REST API. """
+
+        if self.request.version == 'v6':
+            return ScanSerializerV6
+        elif self.request.version == 'v7':
+            return ScanSerializerV6
+
+    def post(self, request, scan_id):
+        try:
+            canceled_ids = Scan.objects.cancel_scan(scan_id)
```

Comment:
I would suggest checking the request version here as well to ensure it's v6 or v7. If not, you'll want to return a 404.